### ISSUE TITLE
Documentation for Klipper users is incorrect

### ISSUE THREAD

**Issue by psiberfunk** [fieldOfView/Cura-LinearAdvanceSettingPlugin#39]
The documentation has an invalid GCODE M900 conversion macro that doesn't work on current versions of klipper (maybe past versions too?). Instead, you should update the readme to this valid alternative:
```

[gcode_macro M900]
gcode:
    # Parameters
    {% set pa = params.K|float %}
    SET_PRESSURE_ADVANCE ADVANCE={pa}
```

**Comment by psiberfunk** [comment]
The existing suggested macro here is incorrect/doesn't comply with klipper's expected syntax:

```
[gcode_macro m900]
default_parameter_K: 0
gcode:
  SET_PRESSURE_ADVANCE ADVANCE={K}

```
Note that when you use the current suggested GCODE macro in printer.cfg in a modern klipper setup, klipper refuses to start with an error.. it doesn't like the line about "default_parameter_K: 0".  I'm not sure if that syntax ever worked properly.  Maybe it was valid in an older version of jinja2 ?

**Comment by fieldOfView** [comment]
See https://github.com/fieldOfView/Cura-LinearAdvanceSettingPlugin/pull/37
I am awaiting confirmation if this fix depends on the version of Klipper being used.

**Comment by drphil3d** [comment]
I'll do you one even better. This allows you to set Pressure Advance and Smooth Time (default is 0.06)

M900 K0 S0.6

[gcode_macro M900]
description: Set Pressure Advance and Smooth Time
gcode:
  SET_PRESSURE_ADVANCE ADVANCE={params.K|default(0)} SMOOTH_TIME={params.S|default(0.06)}

**Comment by fieldOfView** [comment]
I think this has been fixed by #37
